api/internal/features/file-manager/service/manager.go

```diff
@@ -203,20 +213,39 @@ func (f *FileManagerService) MoveDirectory(fromPath string, toPath string) error
 	return nil
 }
 
-// UploadFile handles file upload to the specified path
-func (f *FileManagerService) UploadFile(file io.Reader, path string, filename string) error {
-	f.logger.Log(logger.Info, "uploading file", filename)
+// UploadFile handles file upload to the specified path with support for relative paths in filename
+func (f *FileManagerService) UploadFile(file io.Reader, basePath string, relativePath string) error {
+	f.logger.Log(logger.Info, "uploading file", relativePath)
+
+	// Validate relative path to prevent path traversal attacks
+	if strings.Contains(relativePath, "..") {
```

Comment:
this path traversal detection is performed multiple times across the PR, is there any reason for the same?